tests/testsuite/update.rs

```diff
@@ -1659,3 +1659,56 @@ fn update_with_missing_feature() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_incompatible() {
```

Comment:
We should have a separate dry-run test case that does the same thing

Reply:
I added `--dry-run` as a separate step in the main test. Is that OK?

Reply:
I tend to prefer separate tests. `--dry-run` doesn't have to be as exhaustive.  However, my preference on this isn't too strong, so I leave it to you.